projects/libs/core-library-typescript/src/base-service/service-runner.ts

```diff
@@ -1,27 +1,20 @@
 import { Service } from "./service";
 
 export class ServiceRunner {
-  public static async run(grpcServerFunc: any, service: Service): Promise<void> {
+  public static async run(grpcServerFunc: any, service: Service, endpoint?: string): Promise<void> {
     await service.start();
 
     return grpcServerFunc({
-      endpoint: service.nodeTopology.endpoint,
+      endpoint: endpoint || service.nodeTopology.endpoint,
```

Comment:
@electricmonk why should `endpoint` be optional and otherwise taken from the `service.nodeTopology.endpoint` singleton? Why shouldn't we just require the `endpoint` parameter and avoid accessing the topology here whatsoever?

Reply:
you're right - this is a middle step I did during refactoring to support both approaches, I'll be pushing another commit which removes them